streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -395,11 +397,14 @@ private void maybeUpdateKeyChangingRepartitionNodeMap() {
             final LinkedHashSet<OptimizableRepartitionNode> repartitionNodes = new LinkedHashSet<>();
             for (final StreamsGraphNode keyChangingParent : keyChangingParents) {
                 repartitionNodes.addAll(keyChangingOperationsToOptimizableRepartitionNodes.get(keyChangingParent));
```

Comment:
The stack trace on the ticket indicate that the NPE is thrown in this line, because `keyChangingOperationsToOptimizableRepartitionNodes.get(keyChangingParent)` returns `null` -- can you elaborate why it becomes `null`?

From my current understanding, `keyChangingParents` should be a `Set` and hence we should get each item once and remove it once from `keyChangingOperationsToOptimizableRepartitionNodes` (in the old code). Why would we `get()` the same item twice? Seems I am missing something.